app/common/data/models.py

```diff
@@ -486,6 +486,21 @@ def can_have_child_group(self) -> bool:
         based on the maximum number of levels of nested groups"""
         return bool(self.nested_group_levels < current_app.config["MAX_NESTED_GROUP_LEVELS"])
 
+    @cached_property
+    def contains_add_another_components(self) -> bool:
+        for component in self.cached_all_components:
+            if component.add_another:
+                return True
+        return False
+
+    @cached_property
+    def contains_questions_depended_on_elsewhere(self) -> bool:
+        for component in self.cached_all_components:
+            if len(component.depended_on_by) > 0:
+                return True
```

Comment:
Do these need to be cached? I think we should only cache things that are _needed_, because more caching means more having to remember to cache bust things and potential sources of hard-to-track-down bugs. These are using cached calls so I think, unless these are getting called 100s+ times a request, we should not cache them.

We also have prefixed cached things with `cached_` to help make that clear, so we should do that if we _need_ these to be cached.